sensus/sensus/.ipynb_checkpoints/code-checkpoint.py

```diff
@@ -1,28 +1,91 @@
+### This week was about learning the backpropagation algorithm and implementing the equations in code. I have two short functions left. I will finish those tomorrow and test the network. On Friday I will change the code. Not the content, just the way it is coded. That's because most of it is from a book. I already changed the parts that were meant for Python 2.x. I also changed variable names as some of them did not make sense to me, and almost all the docstring.  
+
+import random 
 import numpy as np
 
 class Network:
     
+    # Initializing the weights and the biases
+    
     def __init__(self, sizes): # 'sizes' is a list, with each element being the number of neurons in a layer
-        self.num_layers = len(sizes)
+        self.n_layers = len(sizes)
         self.sizes = sizes 
-        self.biases = [np.random.randn(j, 1) for j in sizes[1:]] # a vector of initial biases, excluding the input layer
         self.weights = [np.random.randn(j, k) for k, j in zip(sizes[:-1], sizes[1:])] # j and k were swapped deliberately
-        
-    def sigmoid(self, z): 
-        """ The output of a neuron """
-        return 1 / (1 + np.exp(-z)) # the sigmoid function transforms the output into a number between 0 and 1, which becomes
-                                 # input to the next layer
+        self.biases = [np.random.randn(j, 1) for j in sizes[1:]] # a vector of initial biases, excluding the input layer
     
-    def activation(self, a): # a is the vector of outputs from the previous layer (i.e., input of the current layer)
+    # The activation function that computes the input of a neuron as the weighted sum of the previous neurons' outputs, 
+    # the weights and the biases
+    
+    def activation(self, a): # 'a' is a vector of outputs from the previous layer (i.e., input of the current layer)
         """ The activation of a neuron """
```

Comment:
I don't think this is the final version but I just quickly want to add that I think a more elaborate documentation would be helpful, where you also explain what self and a is (instead of doing it after a #). And most of the documentation that I've seen is like this, but I'm not sure if that's the convention:
```suggestion
        """
        The activation of a neuron
        """
```